### PR TITLE
Add Create, Read, Update and Delete Functionalities.

### PR DESCRIPTION
**Pull request by frempongdev**
## In this PR, the following changes were made ♻️ 
- Add Create, Read, Update and Delete Functionalities.
  - Pointed task Array to Local Storage.
  - Added tasks to DOM dynamically.
  - Added Modules for adding tasks.
  - Added Modules for updating tasks.
  - Added Modules for removing tasks.
  - Added appropriate app responses after click.

### CODE REVIEW

File: src/modules/add-task.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  const taskInput = document.getElementById('task-input-field');
+
+  form.addEventListener('submit', (e) => {
+    e.preventDefault();
+    tasks.push({ description: taskInput.value, completed: false, index: tasks.length });
+    form.reset();
+    localStorage.setItem('tasks', JSON.stringify(tasks));
+    displayDom();
+  });
+};
```
Comment (julie-ify) [inline src/modules/add-task.js]: - [x] Each task's index should represent its position in the array. Your tasks start at an index of 0, but they should start from 1. The first task should have an index of 1, the second one 2, the fifth one 5, and so on.

File: src/modules/editTask.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      });
+
+      taskDesc.addEventListener('keyup', (event) => {
+        if (event.keyCode === 13 || event.which === 13) {
+          taskDesc.toggleAttribute('readonly');
+        }
+      });
+    });
+  });
+};
```
Comment (julie-ify) [inline src/modules/editTask.js]: - [x] Your task edit functionality does not work as expected. I am unable to edit a task. After I edit a task it is not reflected in the local storage, so when I refresh the page the old task is what pops up.

- [x] Also after I reload the ability to edit a task (on the page) is gone. Kindly fix that as well.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @frempongdev,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TuyishimireEric — APPROVED** [review]
# Status: APPROVED 🟢  
### Hi  @frempongdev 👋 👋 

#### I'm @TuyishimireEric, Your code reviewer,
Good job so implementing the suggested changes! 🥇 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it 🦾  
Congratulations! 🎉 

## Highlights
✔️professional PR👍 
✔️professional README👍 
✔️No linter Errors 👍 

### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/editTask.js (hunk 2 above)
Comment (frempongdev) [reply src/modules/editTask.js]: All Done.

File: src/modules/add-task.js (hunk 1 above)
Comment (frempongdev) [reply src/modules/add-task.js]: Alright

**Comment by frempongdev** [comment]
Thanks for taking a deep dive into my project, @julie-ify, @TuyishimireEric .

**Pull request merged**